From ecdb59bd71b7406d85fbf5c955921c96b255201b Mon Sep 17 00:00:00 2001
From: Shizhao Liu <lshizhao@amazon.com>
Date: Thu, 27 Feb 2025 14:26:37 -0800
Subject: [PATCH 32/36] Disable MachineSetPreflightChecks Beta feature

MachineSetPreflightChecks feature was promoted from alpha to beta and
is enabled by default, this prevents EKS-A from doing InPlace upgrade
on an cluster whose controlPlane and worker node's major+minor version
are differet. For now we will disable this feature.

Upstream PR: https://github.com/kubernetes-sigs/cluster-api/pull/11228
---
 config/manager/manager.yaml | 2 +-
 feature/feature.go          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/manager/manager.yaml b/config/manager/manager.yaml
index f14cbd505..105fdb920 100644
--- a/config/manager/manager.yaml
+++ b/config/manager/manager.yaml
@@ -23,7 +23,7 @@ spec:
             - "--leader-elect"
             - "--diagnostics-address=${CAPI_DIAGNOSTICS_ADDRESS:=:8443}"
             - "--insecure-diagnostics=${CAPI_INSECURE_DIAGNOSTICS:=false}"
-            - "--feature-gates=MachinePool=${EXP_MACHINE_POOL:=true},ClusterTopology=${CLUSTER_TOPOLOGY:=false},RuntimeSDK=${EXP_RUNTIME_SDK:=false},MachineSetPreflightChecks=${EXP_MACHINE_SET_PREFLIGHT_CHECKS:=true},MachineWaitForVolumeDetachConsiderVolumeAttachments=${EXP_MACHINE_WAITFORVOLUMEDETACH_CONSIDER_VOLUMEATTACHMENTS:=true},PriorityQueue=${EXP_PRIORITY_QUEUE:=false},ReconcilerRateLimiting=${EXP_RECONCILER_RATE_LIMITING:=false},InPlaceUpdates=${EXP_IN_PLACE_UPDATES:=false},MachineTaintPropagation=${EXP_MACHINE_TAINT_PROPAGATION:=false}"
+            - "--feature-gates=MachinePool=${EXP_MACHINE_POOL:=true},ClusterTopology=${CLUSTER_TOPOLOGY:=false},RuntimeSDK=${EXP_RUNTIME_SDK:=false},MachineSetPreflightChecks=${EXP_MACHINE_SET_PREFLIGHT_CHECKS:=false},MachineWaitForVolumeDetachConsiderVolumeAttachments=${EXP_MACHINE_WAITFORVOLUMEDETACH_CONSIDER_VOLUMEATTACHMENTS:=true},PriorityQueue=${EXP_PRIORITY_QUEUE:=false},ReconcilerRateLimiting=${EXP_RECONCILER_RATE_LIMITING:=false},InPlaceUpdates=${EXP_IN_PLACE_UPDATES:=false},MachineTaintPropagation=${EXP_MACHINE_TAINT_PROPAGATION:=false}"
           image: controller:latest
           name: manager
           env:
diff --git a/feature/feature.go b/feature/feature.go
index 5f5956cdc..9ed3258a0 100644
--- a/feature/feature.go
+++ b/feature/feature.go
@@ -94,7 +94,7 @@ func init() {
 var defaultClusterAPIFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	// Every feature should be initiated here:
 	MachinePool:               {Default: true, PreRelease: featuregate.Beta},
-	MachineSetPreflightChecks: {Default: true, PreRelease: featuregate.Beta},
+	MachineSetPreflightChecks: {Default: false, PreRelease: featuregate.Beta},
 	MachineWaitForVolumeDetachConsiderVolumeAttachments: {Default: true, PreRelease: featuregate.Beta},
 	PriorityQueue:                  {Default: false, PreRelease: featuregate.Alpha},
 	ReconcilerRateLimiting:         {Default: false, PreRelease: featuregate.Alpha},
-- 
2.50.1 (Apple Git-155)

